### PR TITLE
DT-1531 Added feature switch for custody updates when we have multipl…

### PR DESCRIPTION
**Pull request by andymarke**
…e custodial events

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/config/FeatureSwitches.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConfigurationProperties(prefix = "features")
+@Data
+public class FeatureSwitches {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/config/FeatureSwitches.java:11]: TODO: I would like to refactor the switches properties to get rid of the nasty nesting. That will be a future PR

File: src/test/java/uk/gov/justice/digital/delius/controller/secure/CustodyKeyDatesControllerTest.java [hunk 2]
```diff
@@ -65,21 +66,24 @@ void setUp() throws ConvictionService.DuplicateActiveCustodialConvictionsExcepti
             when(offenderService.offenderIdOfNomsNumber(any())).thenReturn(Optional.of(99L));
             when(convictionService.getAllActiveCustodialEventsWithBookingNumber(any(), any())).thenReturn(List.of(aCustodyEvent(88L, LocalDate
                 .now())));
-            when(convictionService.addOrReplaceOrDeleteCustodyKeyDates(any(), any(), any())).thenReturn(Custody.builder().sentenceStartDate(LocalDate.now()).build());
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/controller/secure/CustodyKeyDatesControllerTest.java:68]: reformatted file ..

File: src/main/java/uk/gov/justice/digital/delius/service/ConvictionService.java [hunk 3]
```diff
@@ -116,6 +115,7 @@ public ConvictionService(
         this.iapsNotificationService = iapsNotificationService;
         this.contactService = contactService;
         this.telemetryClient = telemetryClient;
+        this.featureSwitches = featureSwitches;
         log.info("NOMIS update custody key dates feature is {}", updateCustodyKeyDatesFeatureSwitch ? "ON" : "OFF");
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/service/ConvictionService.java:119]: log out feature switch value

File: src/testIntegration/resources/application.properties [hunk 4]
```diff
@@ -10,6 +10,9 @@ features.noms.update.custody=true
 features.noms.update.booking.number=true
 features.noms.update.keydates=true
 features.noms.update.noms.number=true
+features.noms.update.multiple-events.update-bulk-key-dates=true
```
Comment (andymarke) [inline src/testIntegration/resources/application.properties:13]: turn off by default

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**